From ff30683418695f5d2cc9e6cf8c9418e09378ebe4 Mon Sep 17 00:00:00 2001
From: Sorah Fukumori <her@sorah.jp>
Date: Tue, 14 Jan 2025 11:01:05 +0900
Subject: [PATCH] agent: Fix a memory leak.

* agent/findkey.c (read_key_file): Free BUF.

--

Cherry pick the master commit of:
	137481fa1002c417cd2c0661b9eefd893b0149d3

Fixes-commit: 434a641d40cbff82beb9f485e0adca72419bfdf2
Signed-off-by: Sorah Fukumori <her@sorah.jp>
---
 agent/findkey.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/agent/findkey.c b/agent/findkey.c
index 1f2938ea3..7a3e90e35 100644
--- a/agent/findkey.c
+++ b/agent/findkey.c
@@ -1182,6 +1182,7 @@ read_key_file (ctrl_t ctrl, const unsigned char *grip,
   }
 
  leave:
+  xfree (buf);
   if (!err && r_keymeta)
     *r_keymeta = pk;
   else
-- 
2.30.2

